src/write.rs

```diff
@@ -458,35 +476,59 @@ where
     fn apply_first(inner: &mut Inner<K, V, M, S>, op: &mut Operation<K, V>) {
         match *op {
             Operation::Replace(ref key, ref mut value) => {
-                let vs = inner.data.entry(key.clone()).or_insert_with(Vec::new);
-                // don't run destructors yet -- still in use by other map
-                for v in vs.drain(..) {
-                    mem::forget(v);
+                let vs = inner.data.entry(key.clone()).or_insert_with(Values::new);
+
+                {
+                    #[cfg(not(feature = "smallvec"))]
+                    let drain = vs.drain(..);
+
+                    #[cfg(feature = "smallvec")]
+                    let drain = vs.drain();
```

Comment:
Same not here about having `let drain;` first, and then do two non-`let` `drain = ` if possible.